pkg/utils/eventrecorder/eventrecorder.go

```diff
@@ -104,24 +118,32 @@ func (e *EventRecorder) findMyNode(ctx context.Context) (corev1.Node, error) {
 	return node, err
 }
 
-// SendNodeEvent sends an event regarding node object
-func (e *EventRecorder) SendNodeEvent(eventType, reason, action, message string) error {
-	// Find my node
-	node, err := e.findMyNode(context.TODO())
+func findMyPod(cachedK8SClient client.Client) (corev1.Pod, error) {
+	var pod corev1.Pod
+	// Find my aws-node pod
+	err := cachedK8SClient.Get(context.TODO(), types.NamespacedName{Name: MyPodName, Namespace: utils.AwsNodeNamespace}, &pod)
 	if err != nil {
-		log.Errorf("Failed to get node: %v", err)
-		return err
+		log.Errorf("Cached client failed GET pod (%s)", MyPodName)
+	} else {
+		log.Debugf("Node found %s - labels - %d", pod.Name, len(pod.Labels))
 	}
+	return pod, err
+}
 
-	// make a copy before modifying the UID
-	// Note: kubectl uses the filter involvedObject.uid=NodeName to fetch the events
-	// that are listed in 'kubectl describe node' output. So setting the node UID to
-	// nodename before sending the event
-	nodeCopy := node.DeepCopy()
-	nodeCopy.SetUID(types.UID(e.HostID))
+// SendNodeEvent sends an event regarding pod object and related to node object
+func (e *EventRecorder) SendNodeEvent(eventType, reason, action, message string) error {
+	instance := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      e.hostNode.Name,
+			Namespace: e.hostNode.Namespace,
+			UID:       types.UID(e.HostID),
+		},
+	}
 
-	e.Recorder.Eventf(nodeCopy, nil, eventType, reason, action, message)
-	log.Debugf("Sent node event: eventType: %s, reason: %s, message: %s, action %s", eventType, reason, message, action)
+	// make the event unique avoid to be aggregated by event broadcaster
+	action = action + ": " + string(uuid.NewUUID())
+	e.Recorder.Eventf(&e.hostPod, instance, eventType, reason, action, message)
```

Comment:
The function name is SendNodeEvent, but this will be pod event right? Am I missing anything?

Reply:
I didn't change the name since the event is still for node label as 
```
new event{
  Regarding `aws-node pod`
  Related to 'node' 
}
```
I have thought if I need rename but was not sure if that is necessary. Thoughts?